pkg/client/api.go

```diff
@@ -81,7 +81,12 @@ func (c *httpAPIClient) Do(ctx context.Context, verb, endpoint string, query url
 	}()
```

Comment:
Let's change the interface so that `Do` always returns an `Error` (*client.Error) instead of the generic `error`. Then, it's much more predictable in the rest of the code that we'd have status code, etc. available